### PR TITLE
Check if template is installed before trying to install it

### PR DESCRIPTION
**Pull request by eaon**
## Description of Changes

Addresses https://github.com/freedomofpress/securedrop-workstation/issues/884#issuecomment-1570597339

This allows an installation to succeed if a template was installed manually even if a template isn't available from `qvm-template` repositories. Applies to both system and `securedrop-workstation-*` templates.

## Testing

- Stage 1, fresh install
   - [ ] Installation succeeds as normal
   - [ ] Running the updater succeeds
- Stage 2, fresh install
  - ~Manually rename `fedora-37` template to a name of your choice, change `sd_supported_fedora_version` to that name~
  - [ ] ~Installation succeeds as normal, despite that template name not referencing available templates in Qubes' or our repositories~
  - [ ] Manual installation of a template from disk (per [#884](https://github.com/freedomofpress/securedrop-workstation/issues/884#issuecomment-1574154240) before SDW installation allows for successful install
### If you have made changes to the provisioning logic

- [ ] All tests (`make test`) pass in `dom0`

### CODE REVIEW

**Comment by rocodes** [comment]
Sorry for the hold-up; it's a small change but I wanted to try out a couple edge cases with how qvm-template reports back just to make sure what we expect of qvm-template is what's happening.

- Template previously installed then removed: `qvm-template` correctly reports it as only `available`
- Template upgraded in place from $oldver to $newver but no changes to template name: `qvm-template info --machine-readable $oldver | grep installed` still shows `oldver` as installed
- Template upgraded in place from `$oldver` to `$newver` and then template renamed to `$newver`: `qvm-template --info` shows _neither_ `$oldver` or `$newver` as installed in the machine-readable output. (This is also true if you rename the template to a random name, see test plan below)
- Template upgraded in place, renamed to random name etc, and then renamed back to original name (`$oldver`) -- `qvm-template info` shows it as `installed` again.
- Template renamed to a random name (test plan):  renamed a fedora template to `fedora-thirtynonsense` then ran `qvm-template info --machine-readable fedora-thirtynonsense` and got `qvm-template: error: No matching templates to list` and a usage message.

So tl;dr people shouldn't be renaming templates or migrating in place, which is good/according to our advice, and if someone upgrades a sys template in place instead of downloading a new one, they shouldn't be able to proceed (which is ideal, also lines up with to our advice- we'd prefer freshly-downloaded sys template to upgraded-in-place ones).  Other than wondering why the discrepancy (between my xp with renaming and the one as described in the test plan), no issues with this change :)

**Review by rocodes — COMMENTED** [review]
(see [comment](https://github.com/freedomofpress/securedrop-workstation/pull/896#issuecomment-1629869551))

**Review by rocodes — APPROVED** [review]
No changes to the PR, but I have modified the test plan to indicate that renaming templates may have unexpected behaviour and is something we should discourage (along with upgrading the templates in place, for our own purposes anyway). These instructions can go into our docs an/or ideally make it into our general installation acceptance testing / QA scenarios.  But otherwise the change wfm and will probably save us a headache or two down the line (like all your work on this project). Thanks @eaon <3  :shipit:
